### PR TITLE
Start phasing out gcloud usages

### PR DESCRIPTION
**Pull request by ahmetb**
Start using client libraries for a few tasks (more to come).

Update dependencies while at it.

cc: @jamesward

### CODE REVIEW

**Comment by ahmetb** [comment]
> Error processing tar file(exit status 1): write /google/devshell/editor/language_servers/java/server/plugins/org.apache.ant_1.10.7.v20190926-0324.jar: no space left on device

The disk space issue is hurting us badly, I suspect we can't escape this anymore. Maybe we should no longer build the docker image.

**Comment by jamesward** [comment]
We could switch GitHub Actions for Cloud Build or some other CI.

File: go.mod [hunk 1]
```diff
@@ -3,8 +3,14 @@ module github.com/GoogleCloudPlatform/cloud-run-button
 go 1.12
 
 require (
-	github.com/AlecAivazis/survey/v2 v2.0.1
-	github.com/briandowns/spinner v0.0.0-20190319032542-ac46072a5a91
-	github.com/fatih/color v1.7.0
-	google.golang.org/api v0.7.1-0.20190723010802-069bea57b1be
+	cloud.google.com/go v0.55.0 // indirect
```
Comment (jamesward) [inline go.mod]: Just curious why these transitive dependencies need to be specified explicitly?

File: cmd/cloudshell_open/billing.go [hunk 2]
```diff
@@ -15,28 +15,22 @@
...
+// GCP project ID.
 func checkBillingEnabled(projectID string) (bool, error) {
-	var o bytes.Buffer
-	var e bytes.Buffer
-	cmd := exec.Command("gcloud", "beta", "billing", "projects", "describe", "-q", "--format=json", projectID)
-	cmd.Stdout = &o
-	cmd.Stderr = &e
-	if err := cmd.Run(); err != nil {
-		return false, fmt.Errorf("error determining if billing account is linked: %+v. output=\n%s", err, e.String())
+	client, err := cloudbilling.NewService(context.TODO())
```
Comment (jamesward) [inline cmd/cloudshell_open/billing.go]: What is `context.TODO()`?

File: go.mod (hunk 1 above)
Comment (ahmetb) [reply go.mod]: Thats what go mod commands currently do. 

They might change their mind later. 
¯\_(ツ)_/¯

File: cmd/cloudshell_open/billing.go (hunk 2 above)
Comment (ahmetb) [reply cmd/cloudshell_open/billing.go]: context is a rather complex topic in Go. It’s used for signaling cancellation in this case. 

Here we are not passing a particular context. So in the future we decide to propagate a context, we’d replace these.

**Comment by ahmetb** [comment]
> We could switch GitHub Actions for Cloud Build or some other CI.

I like the failure emails from Github as well as public failure logs. As far as I know we give up on those when using GCB.

File: cmd/cloudshell_open/billing.go (hunk 2 above)
Comment (jamesward) [reply cmd/cloudshell_open/billing.go]: Ah cool.  Sounds kinda like an alternative to Java's ThreadLocal.  Thanks for the explanation.

**Comment by jamesward** [comment]
> I like the failure emails from Github as well as public failure logs. As far as I know we give up on those when using GCB.

True. We could work together on a GCB for open source solution.  :)

**Comment by ahmetb** [comment]
Superseded by #149.

**Pull request closed without merging**